fs/mmap/fs_mmap.c

```diff
@@ -132,8 +132,7 @@ FAR void *mmap(FAR void *start, size_t length, int prot, int flags,
    * logic is not in place.
    */
 
-  if (prot == PROT_NONE ||
-      (flags & (MAP_FIXED | MAP_DENYWRITE)) != 0)
+  if ((flags & (MAP_FIXED | MAP_DENYWRITE)) != 0)
```

Comment:
can you explain this change?

Reply:
Since the caller may set PROT_NONE initially and then change to PROT_WRITE/PROT_READ... by mprotect later.

Reply:
doesn't mprotect need a permission check then?

Reply:
Yes, but we need save file pointer to check, this will be part of the next patch which not only fix the permission check, but also save the dirty data to the storage.

Reply:
ok